java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -44,7 +50,9 @@
 
   private final OffHeapState offHeap;
   private final DType type;
+  private List<DType> allTypes = new ArrayList<>();
```

Comment:
This is rather confusing without some documentation about what it is and how it is used.  Also how is this supposed to work with structs?  I know that is not a requirement right now, but we should be thinking about it, because it is coming very soon.

Reply:
Thanks Bobby. Yes, it need docs for sure and in my experience this is the harder part of how we want the host side to look like. Without complicating these types further, I think having HostColumnVector deviate from ColumnVector and having some sort of recursive structure may be the way forward to keep it readable and making it work for structs.
For example: 
```HostColumnVector {
childHostColumnVectorMeta[] childMetas;
}
```
 This idea is similar the initial recursive approach we spoke offline about with  a `childHostColumnVectorMeta[]` per `ColumnVector` but we should certainly iron this out as part of this change.